code/app/src/main/java/com/example/snackoverflow/add_ingredient_fragment.java

```diff
@@ -0,0 +1,125 @@
+package com.example.snackoverflow;
+
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.DialogFragment;
+
+import com.google.android.material.textfield.TextInputLayout;
+
+import java.util.Objects;
+
+public class add_ingredient_fragment extends DialogFragment {
+    // Check if we are editing data
+    private boolean edit;
+    // Data storage
+    private TextInputLayout text_input_description;
+    private TextInputLayout text_input_amount;
+    private TextInputLayout text_input_unit;
+    private TextInputLayout text_input_category;
+    // Ingredient object needs to be added
+    private Ingredient ingredient;
+    // listener for Recipe fragment
+    private OnFragmentInteractionListener listener;
+
+    public add_ingredient_fragment(){
+        edit = false;
+    }
+    public add_ingredient_fragment(Ingredient ingredient){
+        this.ingredient = ingredient;
+        edit = true;
+    }
+
+    public interface OnFragmentInteractionListener {
+        void Add_food(Ingredient ingredient);
+        void Edit_food(Ingredient ingredient);
+    }
+
+    @Override
+    public  void onAttach(Context context){
+        // from lab
+        super.onAttach(context);
+        if (context instanceof OnFragmentInteractionListener){
+            listener = (OnFragmentInteractionListener) context;
+        } else {
+            throw new RuntimeException(context.toString()
+                    + " must implement OnFragmentInteractionListener");
+        }
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState){
+        //Inflate the layout
+        View view = LayoutInflater.from(getActivity()).inflate(R.layout.reciepe_ingredients_fragment_layout, null);
+        text_input_description = view.findViewById(R.id.text_input_description);
```

Comment:
Follow variable naming convention to be consistent with others.  Use Camel Case!

Ex: variableName